### ISSUE TITLE
umbContextMenu Directive is not documented

### ISSUE THREAD

**Issue by RachBreeze** [umbraco/UmbracoDocs#2730]
The umbContextMenu directive is not documented in https://our.umbraco.com/apidocs/v8/ui/#/api.

I'm not sure if this is an issue as umbLogin isn't either but navResize is (they are all defined as application-level directives).

**Comment by sofietoft** [comment]
Thanks for raising this @RachBreeze !

As I've understood, the Angular docs are auto-generated - strange if something is missing 🤔 
I'll forward it to our developers, so we can figure this one out!

**Comment by sofietoft** [comment]
@RachBreeze - I have an update on this one.
The reason that the `umbContextMenu` is not documented on https://our.umbraco.com/apidocs/v8/ui/#/api, is because the code doesn't have any ngdocs comments associated with it. 

These comments will need to be added in the source code - for `umbContextMenu` that would be here: https://github.com/umbraco/Umbraco-CMS/blob/v8/contrib/src/Umbraco.Web.UI.Client/src/common/directives/components/application/umbcontextmenu.directive.js

As a reference, have a look at the source code for the `navResize` directive, which does include comments: https://github.com/umbraco/Umbraco-CMS/blob/v8/contrib/src/Umbraco.Web.UI.Client/src/common/directives/components/application/navresize.directive.js

So basically, the Angular docs are generated based on the comments added to the source code.

Additionally, work is being put into adding some angular docs to Our as well. Have a look at this issue: https://github.com/umbraco/UmbracoDocs/issues/2653 - perhaps we should add `umbContextMenu` and `umbLogin` as well?

**Comment by RachBreeze** [comment]
Thank you I will take a look at adding the comments to umbContextMenu

**Comment by sofietoft** [comment]
Related PR merged.
I'll close this issue as well 💪
